### PR TITLE
MusicXML: Add support for part-name-display

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR adds import/export of MusicXML's [`part-name-display`](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/part-name-display/).

fixes https://musescore.org/en/node/330480

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
I believe ypu can rebase and try again now

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -7297,21 +7325,29 @@ static void partList(XmlWriter& xml, Score* score, MxmlInstrumentMap& instrMap)
...
-                xml.tag("part-name", "");
+        String partName = part->longName();
+        // use the track name if no part long name
+        if (partName.empty()) {
+            partName = part->partName();
+            if (!partName.empty()) {
+                attributes.push_back({ "print-object", "no" });
             }
         }
+        xml.tag("part-name", attributes, MScoreTextToMXML::toPlainText(partName).replace(u"♭", u"b"));
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:7338]: I know that it is quite unlikely to happen, but why not checking for sharp and replacing that too, esp. as ?writeDisplayName()`supports that?

File: src/importexport/musicxml/tests/data/testPartNames2.xml [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+        <display-text> Bass</display-text>
+        </part-name-display>
+      <part-abbreviation>Eb Bs.</part-abbreviation>
+      <part-abbreviation-display>
+        <display-text>E</display-text>
+        <accidental-text>flat</accidental-text>
+        <display-text> Bs.</display-text>
+        </part-abbreviation-display>
+      <score-instrument id="P1-I1">
+        <instrument-name>Eb Tuba (Treble Clef)</instrument-name>
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/tests/data/testPartNames2.xml:37]: Why not
```xml
        <instrument-name>E♭ Tuba (Treble Clef)</instrument-name>
```

File: src/importexport/musicxml/tests/data/testPartNames2.xml [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+        <display-text> Bass</display-text>
+        </part-name-display>
+      <part-abbreviation>Bb Bs.</part-abbreviation>
+      <part-abbreviation-display>
+        <display-text>B</display-text>
+        <accidental-text>flat</accidental-text>
+        <display-text> Bs.</display-text>
+        </part-abbreviation-display>
+      <score-instrument id="P2-I1">
+        <instrument-name>Bb Tuba (Treble Clef)</instrument-name>
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/tests/data/testPartNames2.xml:61]: likewise

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 4]
```diff
@@ -7335,7 +7335,7 @@ static void partList(XmlWriter& xml, Score* score, MxmlInstrumentMap& instrMap)
                 attributes.push_back({ "print-object", "no" });
             }
         }
-        xml.tag("part-name", attributes, MScoreTextToMXML::toPlainText(partName).replace(u"♭", u"b"));
+        xml.tag("part-name", attributes, MScoreTextToMXML::toPlainText(partName));
         if (partName.contains(u"♭")) {
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:7339]: here's you may want to check for `u"♯"`too
As well as further down.

File: src/importexport/musicxml/internal/musicxml/musicxmlfonthandler.cpp [hunk 5]
```diff
@@ -68,7 +68,7 @@ String MScoreTextToMXML::toPlainText(const String& text)
         }
     }
     //LOGD("MScoreTextToMXML::toPlainText('%s') res '%s'", muPrintable(text), muPrintable(res));
-    return res;
+    return res.replace(u"♭", u"b").replace(u"♯", u"#");
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/musicxmlfonthandler.cpp:71]: It is not, it changes instrument names on MusicXML files too. No need to as far as I can tell

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/musicxmlfonthandler.cpp:71]: And it's purpose is entirely different: it is to remove HTML(-like) tags

[the remark this answers is not in the document]
Comment (miiizen) [reply src/importexport/musicxml/internal/musicxml/musicxmlfonthandler.cpp:71]: Yes, we'd prefer to keep the behaviour of this to simply stripping tags from text without any other processing.

**Review by miiizen — APPROVED** [review]

**Pull request merged**